functional-tests/plege_sector_test.go

```diff
@@ -0,0 +1,98 @@
+package functional
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/internal/app/go-filecoin/node"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/clock"
+	tf "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers/testflags"
+	gengen "github.com/filecoin-project/go-filecoin/tools/gengen/util"
+)
+
+func TestMiningPledgeSector(t *testing.T) {
+	tf.FunctionalTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	wd, _ := os.Getwd()
+	genCfgPath := filepath.Join(wd, "..", "fixtures/setup.json")
+	presealPath := filepath.Join(wd, "..", "fixtures/genesis-sectors")
+	genTime := int64(1000000000)
+	blockTime := 1 * time.Second
+	fakeClock := clock.NewFake(time.Unix(genTime, 0))
+
+	// Load genesis config fixture.
+	genCfg := loadGenesisConfig(t, genCfgPath)
+	genCfg.Miners = append(genCfg.Miners, &gengen.CreateStorageMinerConfig{
+		Owner:      1,
+		SectorSize: 2048,
+	})
+	seed := node.MakeChainSeed(t, genCfg)
+	chainClock := clock.NewChainClockFromClock(uint64(genTime), blockTime, fakeClock)
+
+	bootstrapMiner := makeNode(ctx, t, seed, chainClock)
+	_, _, err := initNodeGenesisMiner(t, bootstrapMiner, seed, genCfg.Miners[0].Owner, presealPath, genCfg.Miners[0].SectorSize)
+	require.NoError(t, err)
+
+	newMiner := makeNode(ctx, t, seed, chainClock)
+	seed.GiveKey(t, newMiner, 1)
+	_, _ = seed.GiveMiner(t, newMiner, 1)
+
+	err = bootstrapMiner.Start(ctx)
+	require.NoError(t, err)
+	err = newMiner.Start(ctx)
+	require.NoError(t, err)
+	defer bootstrapMiner.Stop(ctx)
+	defer newMiner.Stop(ctx)
+
+	node.ConnectNodes(t, newMiner, bootstrapMiner)
+
+	// Have bootstrap miner mine continuously so newMiner's pledgeSector can put multiple messages on chain.
+	go simulateBlockMining(ctx, t, fakeClock, blockTime, bootstrapMiner)
+
+	err = newMiner.StorageMining.Start(ctx)
+	require.NoError(t, err)
+
+	err = newMiner.PieceManager().PledgeSector(ctx)
+	require.NoError(t, err)
+
+	// wait while checking to see if the new miner has added any sectors (indicating sealing was successful)
+	for i := 0; i < 100; i++ {
+		ts, err := newMiner.PorcelainAPI.ChainHead()
+		require.NoError(t, err)
+
+		maddr, err := newMiner.BlockMining.BlockMiningAPI.MinerAddress()
+		require.NoError(t, err)
+
+		status, err := newMiner.PorcelainAPI.MinerGetStatus(ctx, maddr, ts.Key())
+		require.NoError(t, err)
+
+		if status.SectorCount > 0 {
+			return
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+	t.Fatal("Did not add sectors in the allotted time")
+}
+
+func simulateBlockMining(ctx context.Context, t *testing.T, fakeClock clock.Fake, blockTime time.Duration, node *node.Node) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			fakeClock.Advance(blockTime)
+			_, err := node.BlockMining.BlockMiningAPI.MiningOnce(ctx)
+			require.NoError(t, err)
+		}
+		time.Sleep(time.Second)
```

Comment:
Why do you need to sleep? If the fake clock is working properly there should be no need to sleep here. This could be more obvious by setting the blockTIme to be high.

A miniscule sleep might be necessary for chain sync to process the new block.

Reply:
It's not necessary. Removed.